### DISCUSSION TITLE
Mirage | Upgrade v13.4.1 to v16.4.1 bugs

### DISCUSSION THREAD

**Discussion by jncalderon** [primefaces/community#130] [PrimeNG Templates] [answered]
Hi just upgrade v13.4.1 to v16.4.1. Everything looks good, few design bugs that would be easy to fix. The main issues is on some components that design is bad.
I'm using Mirage template (Also updated it to v16.0.0).

My template is very clean, I didn't customize layout and themes, only some colors. Some upgrade guide or considerations for my context upgrade?

![image](https://github.com/primefaces/community/assets/15347917/6d38ca8f-fe15-499d-a5e2-754bd81a2b79)
![image](https://github.com/primefaces/community/assets/15347917/835c2352-fa65-4037-9feb-624289d5b76f)
![image](https://github.com/primefaces/community/assets/15347917/590ec30b-df23-4bb4-b4cd-d7d368d66761)
![image](https://github.com/primefaces/community/assets/15347917/f320cf24-ff4e-4e97-94f0-241bc1bef7eb)

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

This issue happens because templates are using primeng 16.0.2 and primeng-sass-theme 16.0.2. Since we're currently implementing accessibility to PrimeNG, we've updated a lot of styles between 16.0.2 and 16.4.1. To solve this issue manually without waiting us to update all templates, do the steps below;

1-) Clone primeng-sass-theme repo on your local machine https://github.com/primefaces/primeng-sass-theme
2-) Go to primefaces store and download v16 of mirage-ng
3-) Take the assets and theme folder of v16 and replace it with your own
4-) open the primeng-sass-theme, take the theme-base folder and replace it with your own in `src/assets/sass/theme folder`
5-) In primeng-sass-theme open the `themes/bootstrap4/bootstrap4-dark/_variables.scss`,  copy all the things below line 28, paste it into `_variables_dark.scss` file in your project. Located in `src/assets/sass/theme` folder
6-) do the same step for `themes/bootstrap4/boostrap4-light/_variables.scss` copy all the things below line 28 and paste it into `_variables_light.scss` file in your project.
7-) run `sass --update src/assets:src/assets --no-source-map` command
8-) serve the project by `ng serve`

After the steps above, your project will be correctly updated to the latest primeng-sass-theme and primeng.

I'm updating the category to PrimeNG Templates and refactoring the topic since it's related to MirageNG Template.
Please let us know if anything goes wrong.

**Comment by jncalderon** [comment]
thanks a lot!! I following the steps and now is working very well, thanks!! I going to keep on the loop to get the updates

**Comment by jncalderon** [comment]
Just now I have this error, but I going to follow other threds , It's a bug that someone else is working on it
![image](https://github.com/primefaces/community/assets/15347917/88b9baa8-339d-40ec-bce4-7919e443a93f)

**Reply by cetincakiroglu** [reply]
Hi, we've merged the fix for the issue, it'll be available on upcoming release https://github.com/primefaces/primeng/pull/13644

**Comment by admt1** [comment]
I have this problem with Saga Blue, how can i fix it? Or when will be fix by you?

**Comment by dnordahl1** [comment]
In the case of the site I'm working on, currently it has a css files in a parent project in relation to the Angular 2 portion, including bootstrap.css (a setup I inherited to maintain, and I'm looking to address in the near future).  That css is referenced by a parent content page that references this file in a _Layout.cshtml as:

`<link href="~/css/bootstrap.css" rel="stylesheet">`

So to not edit the complied bootstrap.css, I created a "wrapper" css file called bootstrap_layer_wrapper.css containing:

` @import "bootstrap.css" layer(bootstrap);`

and changed the link in the _Layout.cshtml to point to that file instead of bootstrap.css.  Then in the Angular 2 project's styles.scss I have: 

`@layer bootstrap, base, primeng, app;` 

Btw, I feel it's extremely inconsiderate to have a change this fundamentally breaking not only introduced on a minor version, but also not documented at all on the release notes or marked as breaking with the package libraries after so many releases. So much time wasted by users having to figure out the cause of their problems on their own, and so many more releases and time has passed and still nothing to address this or give people a heads up about it in any form.  Related.. why have upgrade/migration guides and videos stopped being provided after version 12?  That is another way this could have been communicated.  

And what about this glaring bug which basically makes any version of primeNg 16 unusable if a site has tables with context menus goes unaddressed after 7 minor releases now:  https://github.com/primefaces/primeng/issues/13934  This immediately breaks several of our e2e tests on updating to any version of primeNg 16.  How were so many releases put out without this being noticed by your automated tests, and is this not an extremely critical bug to not either fix or provide a workaround for in the mean time? We can't update to 16 with this going on with context menus.

**Comment by AlbaSS18** [comment]
Hi, does anyone know if this is fixed in v17? I have just updated and input component still has this style.
![imagen](https://github.com/primefaces/community/assets/46462037/7e6d346d-c3e4-4c6e-8150-dfc2624bf0f9)
